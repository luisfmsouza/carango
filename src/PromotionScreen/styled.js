import styled from "styled-components";

import colors from "../theme/colors";

export const Page = styled.div`
  background-color: ${colors.blue10};
`;

export const Main = styled.div`
  background-color: ${colors.blue10};
  display: grid;
  grid-template-columns: 1fr;
  grid-column-gap: 35px;
  padding: 60px 25px;

  @media (min-width: 600px) {
    grid-template-columns: 1fr 1fr;
  }
`;

export const Column = styled.section``;

export const Title = styled.h1`
  color: ${colors.blue20};
  font-family: LeasePlan, Arial, Helvetica, sans-serif;
  font-size: 36px;
  line-height: 1.2;
  margin: 22px 0;
`;

export const Paragraph = styled.p`
  color: ${colors.grey20};
  font-family: LeasePlan, Arial, Helvetica, sans-serif;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5;
`;
