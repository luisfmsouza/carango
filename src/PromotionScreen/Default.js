import React from "react";

import Form from "../components/Form";
import Header from "../components/Header";

import { Page, Main, Column, Title, Paragraph } from "./styled";

export default ({
  formState,
  shouldBlockFormSubmission,
  handleInputChange,
  handleInputBlur,
  handleSubmit
}) => {
  return (
    <Page>
      <Header />
      <Main>
        <Column>
          <Title>Get a free year of driving!</Title>
          <Paragraph>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat
            eros, finibus ac ullamcorper sit amet, tempus quis lacus. Quisque
            eleifend semper pretium. Sed rutrum odio lectus, id maximus nunc
            rhoncus sed. Praesent quis suscipit dui. Lorem ipsum dolor sit amet,
            consectetur adipiscing elit. Integer orci dui, rutrum eu feugiat
            lobortis, tristique sollicitudin est. Nulla tincidunt semper nibh at
            pulvinar. Etiam tempor quam vitae neque ornare efficitur. Morbi id
            egestas est, at viverra orci. Fusce eleifend placerat ligula, eget
            viverra lectus facilisis vitae. Praesent placerat tortor a mi
            laoreet, quis lobortis nunc eleifend.
          </Paragraph>
        </Column>

        <Column>
          <Title>Get a car leasing offer</Title>
          <Paragraph>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat
            eros, finibus ac ullamcorper sit amet, tempus quis lacus.
          </Paragraph>
          <Form
            formState={formState}
            shouldBlockFormSubmission={shouldBlockFormSubmission}
            handleInputChange={handleInputChange}
            handleInputBlur={handleInputBlur}
            handleSubmit={handleSubmit}
          />
        </Column>
      </Main>
    </Page>
  );
};
