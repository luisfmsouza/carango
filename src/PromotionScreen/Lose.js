import React from "react";

import Header from "../components/Header";

import { Page, Main, Column, Title, Paragraph } from "./styled";

export default () => {
  return (
    <Page>
      <Header />
      <Main>
        <Column>
          <Title>Better luck next time!</Title>
          <Paragraph>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat
            eros, finibus ac ullamcorper sit amet, tempus quis lacus. Quisque
            eleifend semper pretium.
          </Paragraph>
        </Column>
      </Main>
    </Page>
  );
};
