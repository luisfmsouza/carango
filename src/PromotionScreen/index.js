import React from "react";

import Win from "./Win";
import Lose from "./Lose";
import Default from "./Default";

export default ({ screen }) => {
  if (screen === "win") return <Win />;
  if (screen === "lose") return <Lose />;

  return <Default />;
};
