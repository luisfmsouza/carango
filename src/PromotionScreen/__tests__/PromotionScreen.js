import React from "react";
import renderer from "react-test-renderer";

import PromotionScreen from "..";

describe("PromotionScreen", () => {
  it("renders the PromotionScreen default component", () => {
    const actual = renderer.create(<PromotionScreen />).toJSON();

    expect(actual).toMatchSnapshot();
  });

  it("renders the PromotionScreen component when win the prize", () => {
    const actual = renderer.create(<PromotionScreen screen="win" />).toJSON();

    expect(actual).toMatchSnapshot();
  });

  it("renders the PromotionScreen component when lose the prize", () => {
    const actual = renderer.create(<PromotionScreen screen="lose" />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
