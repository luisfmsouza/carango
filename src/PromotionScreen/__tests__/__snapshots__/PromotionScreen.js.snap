// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionScreen renders the PromotionScreen component when lose the prize 1`] = `
<div
  className="sc-htpNat lftjIv"
>
  <header
    className="sc-bdVaJa blMrvh"
  >
    <a
      href="#"
    >
      <img
        alt="CarNext logo"
        className="sc-bwzfXH hqJQNA"
        src="car-next-logo.png"
        title="CarNext logo"
      />
    </a>
  </header>
  <div
    className="sc-bxivhb hdqdbk"
  >
    <section
      className="sc-ifAKCX cFlEyZ"
    >
      <h1
        className="sc-EHOje fvXwPD"
      >
        Better luck next time!
      </h1>
      <p
        className="sc-bZQynM GNZEL"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat eros, finibus ac ullamcorper sit amet, tempus quis lacus. Quisque eleifend semper pretium.
      </p>
    </section>
  </div>
</div>
`;

exports[`PromotionScreen renders the PromotionScreen component when win the prize 1`] = `
<div
  className="sc-htpNat lftjIv"
>
  <header
    className="sc-bdVaJa blMrvh"
  >
    <a
      href="#"
    >
      <img
        alt="CarNext logo"
        className="sc-bwzfXH hqJQNA"
        src="car-next-logo.png"
        title="CarNext logo"
      />
    </a>
  </header>
  <div
    className="sc-bxivhb hdqdbk"
  >
    <section
      className="sc-ifAKCX cFlEyZ"
    >
      <h1
        className="sc-EHOje fvXwPD"
      >
        Congratulations, you won!
      </h1>
      <p
        className="sc-bZQynM GNZEL"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat eros, finibus ac ullamcorper sit amet, tempus quis lacus. Quisque eleifend semper pretium.
      </p>
    </section>
  </div>
</div>
`;

exports[`PromotionScreen renders the PromotionScreen default component 1`] = `
<div
  className="sc-htpNat lftjIv"
>
  <header
    className="sc-bdVaJa blMrvh"
  >
    <a
      href="#"
    >
      <img
        alt="CarNext logo"
        className="sc-bwzfXH hqJQNA"
        src="car-next-logo.png"
        title="CarNext logo"
      />
    </a>
  </header>
  <div
    className="sc-bxivhb hdqdbk"
  >
    <section
      className="sc-ifAKCX cFlEyZ"
    >
      <h1
        className="sc-EHOje fvXwPD"
      >
        Get a free year of driving!
      </h1>
      <p
        className="sc-bZQynM GNZEL"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat eros, finibus ac ullamcorper sit amet, tempus quis lacus. Quisque eleifend semper pretium. Sed rutrum odio lectus, id maximus nunc rhoncus sed. Praesent quis suscipit dui. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer orci dui, rutrum eu feugiat lobortis, tristique sollicitudin est. Nulla tincidunt semper nibh at pulvinar. Etiam tempor quam vitae neque ornare efficitur. Morbi id egestas est, at viverra orci. Fusce eleifend placerat ligula, eget viverra lectus facilisis vitae. Praesent placerat tortor a mi laoreet, quis lobortis nunc eleifend.
      </p>
    </section>
    <section
      className="sc-ifAKCX cFlEyZ"
    >
      <h1
        className="sc-EHOje fvXwPD"
      >
        Get a car leasing offer
      </h1>
      <p
        className="sc-bZQynM GNZEL"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In erat eros, finibus ac ullamcorper sit amet, tempus quis lacus.
      </p>
      <form
        className="sc-gzVnrw ibZhkv"
      >
        <div
          className="sc-htoDjs hZvHaw"
        >
          <div
            className="sc-dnqmqq hlbUkZ"
          >
            <label
              className="sc-iwsKbI dOiXxC"
              htmlFor="form--first-name"
            >
              First name
            </label>
            <input
              className="sc-gZMcBi gJdFSv"
              id="form--first-name"
              placeholder="Enter first name"
              type="text"
            />
          </div>
          <div
            className="sc-dnqmqq hlbUkZ"
          >
            <label
              className="sc-iwsKbI dOiXxC"
              htmlFor="form--surname"
            >
              Surname
            </label>
            <input
              className="sc-gZMcBi gJdFSv"
              id="form--surname"
              placeholder="Enter surname"
              type="text"
            />
          </div>
        </div>
        <div
          className="sc-dnqmqq hlbUkZ"
        >
          <label
            className="sc-iwsKbI dOiXxC"
            htmlFor="form--email"
          >
            Email address
          </label>
          <input
            className="sc-gZMcBi gJdFSv"
            id="form--email"
            placeholder="Enter your email address"
            type="email"
          />
        </div>
        <div
          className="sc-dnqmqq hlbUkZ"
        >
          <label
            className="sc-iwsKbI dOiXxC"
            htmlFor="form--phone"
          >
            Phone number
          </label>
          <input
            className="sc-gZMcBi gJdFSv"
            id="form--phone"
            placeholder="Enter your phone number"
            type="tel"
          />
        </div>
        <div
          className="sc-dnqmqq hlbUkZ"
        >
          <label
            className="sc-iwsKbI dOiXxC"
            htmlFor="form--voucher"
          >
            Voucher code
          </label>
          <input
            className="sc-gZMcBi gJdFSv"
            id="form--voucher"
            placeholder="Enter your voucher code"
            type="number"
          />
        </div>
        <div
          className="sc-dnqmqq hlbUkZ"
        >
          <input
            className="sc-VigVT juvwGh"
            id="form--policy"
            type="checkbox"
          />
          <span
            className="sc-jTzLTM dnRskf"
          />
          <label
            className="sc-gqjmRU lbtNyB"
            htmlFor="form--policy"
          >
            Accept Privacy Settlement
          </label>
        </div>
        <button
          className="sc-fjdhpX cuAGuH"
          type="submit"
        >
          <span
            className="sc-jzJRlG iqolyC"
          >
            Get Offer
          </span>
          <svg
            className="sc-cSHVUG hVuVMI"
          >
            arrow-right.svg
          </svg>
        </button>
      </form>
    </section>
  </div>
</div>
`;
